### PR TITLE
[dagster-aws] [docs] add docs for PipesEMRClient

### PR DESCRIPTION
**Pull request by danielgafni**
## Summary & Motivation

Adding API docs & tutorial for AWS EMR Pipes

### CODE REVIEW

**Comment by danielgafni** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25011?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25011** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25011?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23998** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23998?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24884** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24884?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24469** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24469?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#24809](https://github.com/dagster-io/dagster/pull/24809) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24809?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#24968** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24968?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24948** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24948?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danielgafni and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/emr_serverless.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def run(
         Args:
             context (OpExecutionContext): The context of the currently executing Dagster op or asset.
             params (dict): Parameters for the ``start_job_run`` boto3 AWS EMR Serverless client call.
-                See `Boto3 API Documentation <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/emr-serverless/client/start_job_run.html>`_
+                See `Boto3 EMR Serverless API Documentation <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/emr-serverless/client/start_job_run.html>`_
```
Comment (danielgafni) [inline python_modules/libraries/dagster-aws/dagster_aws/pipes/clients/emr_serverless.py:96]: sphinx was complaining about this entry being identical to another one so I decided to inject service name here

File: docs/sphinx/sections/api/apidocs/libraries/dagster-aws.rst [hunk 2]
```diff
@@ -122,6 +122,10 @@ Clients
 
 .. autoclass:: dagster_aws.pipes.PipesECSClient
 
+.. autoclass:: dagster_aws.pipes.PipesEMRClient
+
+.. autoclass:: dagster_aws.pipes.PipesEMRServerlessClient
```
Comment (danielgafni) [inline docs/sphinx/sections/api/apidocs/libraries/dagster-aws.rst:127]: Oops... was missing

File: docs/content/concepts/dagster-pipes/aws-emr.mdx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+This tutorial gives a short overview on how to use [Dagster Pipes](/concepts/dagster-pipes) with [AWS EMR](https://aws.amazon.com/emr/).
+
+The [dagster-aws](/\_apidocs/libraries/dagster-aws) integration library provides the <PyObject object="PipesEMRClient" module="dagster_aws.pipes" /> resource, which can be used to launch AWS EMR jobs from Dagster assets and ops. Dagster can receive regular events such as logs, asset checks, or asset materializations from jobs launched with this client. Using it requires minimal code changes to your EMR jobs.
+
+---
+
+## Prerequisites
+
+- **In the orchestration environment**, you'll need to:
```
Comment (cmpadden) [inline docs/content/concepts/dagster-pipes/aws-emr.mdx:16]: Referring this to the "the orchestration environment" had cognitive overhead, where I had to think what you meant by this, when I think we could just say "In your Dagster environment install `dagster-aws`.

File: docs/content/concepts/dagster-pipes/aws-emr.mdx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+  - **AWS authentication credentials configured.** If you don't have this set up already, refer to the [boto3 quickstart](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html).
+
+- **In AWS**:
+
+  - An existing AWS account
+  - Prepared infrastructure such as S3 buckets, IAM roles, and other resources required for your EMR job
+
+---
+
+## Step 1: Install the dagster-pipes module
```
Comment (cmpadden) [inline docs/content/concepts/dagster-pipes/aws-emr.mdx:35]: Can we be explicit that this is installing it in the EMR environment?
```suggestion
## Step 1: Install the dagster-pipes module in your EMR environment
```

File: docs/content/concepts/dagster-pipes/aws-emr.mdx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+COPY --from=builder /output/venv.pex /venv.pex
+```
+
+The build can be launched with:
+
+```shell
+DOCKER_BUILDKIT=1 docker build --output type=local,dest=./output .
+```
+
+Then, upload the `output` directory to an S3 bucket and use the `.pex` file with `spark-submit` in your EMR step:
```
Comment (cmpadden) [inline docs/content/concepts/dagster-pipes/aws-emr.mdx:78]: Perhaps we show an example `aws s3 sync` or `cp` command here?

**Review by cmpadden — CHANGES_REQUESTED** [review]
Overall this looks great--just a couple of small comments.

File: docs/content/concepts/dagster-pipes/aws-emr.mdx (hunk 3 above)
Comment (danielgafni) [reply docs/content/concepts/dagster-pipes/aws-emr.mdx:16]: ok made this change in all Pipes tutorials

File: docs/content/concepts/dagster-pipes/aws-emr.mdx (hunk 4 above)
Comment (danielgafni) [reply docs/content/concepts/dagster-pipes/aws-emr.mdx:35]: same, fixed in all tutorials

File: docs/content/concepts/dagster-pipes/aws-emr.mdx (hunk 5 above)
Comment (danielgafni) [reply docs/content/concepts/dagster-pipes/aws-emr.mdx:78]: done

**Comment by danielgafni** [comment]
### Merge activity

* **Oct 18, 7:11 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/25011) disabled "merge when ready" on this PR due to: a merge conflict with the target branch; resolve the conflict and try again..

**Review by cmpadden — APPROVED** [review]

**Pull request merged**